docs/articles/documentation/reference/test-api/selector/withtext.md

```diff
@@ -0,0 +1,69 @@
+---
+layout: docs
+title: Selector.withText Method
+permalink: /documentation/reference/test-api/selector/withtext.html
+---
+# Selector.withText Method
+
+Finds an element whose content includes the specified text.
+
+## Syntax
+
+### withText(text)
+
+```text
+Selector().withText(text) → Selector
+```
+
+Selects elements that *contain* the specified text.
+
+Argument | Type   | Description
+-------- | ------ | --------------
+`text`  | String | The element's text content. The `text` argument is case-sensitive.
+
+To filter elements by *strict match*, use the [withExactText](withexacttext.md) method.
+
+### withText(re)
+
+```text
+Selector().withText(re) → Selector
+```
+
+Selects elements whose text content matches the specified regular expression.
+
+Argument | Type   | Description
+-------- | ------ | --------------
+`re`  | RegExp | A regular expression that should match the element's text.
+
+## Examples
+
+```js
+// Selects label elements that contain 'foo'.
+// Matches 'foo', 'foobar'. Does not match 'bar', 'Foo'.
+Selector('label').withText('foo');
+
+// Selects div elements whose text matches
+// the /a[b-e]/ regular expression.
+// Matches 'ab', 'ac'. Does not match 'bb', 'aa'.
+Selector('div').withText(/a[b-e]/);
+```
+
+## Notes
+
+`withText` selects the element that immediately contains the specified text and its ancestors.
```

Comment:
"that immediately contains" How can something 'immediately' contain something?

Reply:
*`withText` selects the element that contains the specified text and its ancestors.*

Reply:
Ok